### DISCUSSION TITLE
GLB PBR Runtime Export

### DISCUSSION THREAD

**Discussion by EricBeetsOfficial-Opuscope** [atteneder/glTFast#344] [Q&A]
Hi Andreas,

I read the runtime export is still experimental (only albedo and occlusion textures seem to be supported).
Is there a plan to support the fully PBR material for a runtime export (bump seems to be easy to add, but metallic-roughness seems harder ..) ?
Do you have special instructions to try working on a PR for it ?

Thanks in advance,
Regards,
Eric

**Comment by atteneder** [comment]
Hi @EricBeetsOfficial-Opuscope ,

The state of export (what is supported and what not) can be found in the [documentation->features->Material Export](https://github.com/atteneder/glTFast/blob/main/Documentation%7E/features.md#material-export) and currently in some more detail in #260 .

Support for more PBR features, including occlusion-roughness-metallness, was added with 4.6.0.

I still consider export experimental, as it was only tested with the most important shaders of URP/HDRP (Lit mainly).

hth

**Comment by EricBeetsOfficial-Opuscope** [comment]
Oh .. I probably didn't understand the runtime export.
It supports the Unity Built-in `Universal Render Pipeline/Lit` shader, but not the glTFast `Shader Graphs/glTF-pbrMetallicRoughness` ?

I can reproduce the behaviour:
- Import the DamagedHelmet (GLTF format) in Editor (from https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/DamagedHelmet/glTF)
- Drag and drop into my scene (with the correct shader `Shader Graphs/glTF-pbrMetallicRoughness-Opaque`)
- Export this model (right click,  in the hierarchy `Export GLTF`-> `glTF (.gltf)`).
- In the finder I only see the albedo texture (there is the same behaviour in runtime).

Indeed, if I switch to the built-in PBR and re-switch to the glTFast PBR, the exporter save the albedo/occlusion/emissive/normal texture.. is that a bug ? Or I maybe do something wrong ?

**Reply by atteneder** [reply]
It's more of a missing feature than a bug. Re-exporting previously imported glTF is not as big of a use-case than exporting existing Unity assets (using default Standard/Lit shaders).

**Reply by EricBeetsOfficial-Opuscope** [reply]
oh ok .. we actually wanted to be able to export a glb from an imported gltf (using glTFast as a runtime converter)

**Reply by atteneder** [reply]
@EricBeetsOfficial-Opuscope I see. Out of curiosity, what is your use-case? I guess you/the user edits the scene in some way, right?

**Reply by EricBeetsOfficial-Opuscope** [reply]
Actually, users can import models (from an url, it could be glb or gltf files). Then, we store the scene on our servers (to be able to re-open the composed scene later). It would easier for us to store a unique glb file by model (instead of a .zip or a gltf with resources files). That's why we would need a runtime converter from gltf to glb 😉

**Reply by atteneder** [reply]
fyI: Opened #351

**Reply by EricBeetsOfficial-Opuscope** [reply]
oh great ! thank you !!
